pkg/txn/client/entire_txn_client.go

```diff
@@ -0,0 +1,68 @@
+// Copyright 2022 Matrix Origin
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+// EntireClient is a wrapper to support txn for temp engine.
+type EntireClient struct {
```

Comment:
Since there are actually two sets of engines after the temporary table is created, that means the transaction may be cross-engine. 

Such a transaction may correspond to multiple TXnoperators, and each txnop must be called separately for commit and rollback

Reply:
So like Engine, Operator and Client also need to wrap a layer.